### PR TITLE
Create a pro engine connector hook in Naming_AST

### PR DESCRIPTION
**Pull request by ihji**

### CODE REVIEW

File: src/naming/Naming_AST.ml [hunk 1]
```diff
@@ -473,497 +473,494 @@ let assign_implicitly_declares lang =
   lang =*= Lang.Python || lang =*= Lang.Ruby || lang =*= Lang.Php
   || Lang.is_js lang
 
+class virtual ['self] default_visitor =
```
Comment (IagoAbal) [inline src/naming/Naming_AST.ml:476]: This diff is huge something looks wrong.

File: src/naming/Naming_AST.mli [hunk 2]
```diff
@@ -1,5 +0,0 @@
-(* Works by side effect on the generic AST by modifying its refs.
```
Comment (IagoAbal) [inline src/naming/Naming_AST.mli:1]: This looks fishy too...

File: src/analyzing/Constant_propagation.ml [hunk 3]
```diff
@@ -642,6 +642,17 @@ let propagate_basic lang prog =
       (* the defs *)
       method! visit_definition (env, ctx) x =
         match x with
+        | ( {
```
Comment (IagoAbal) [inline src/analyzing/Constant_propagation.ml:645]: Wouldn't this be the thing that we add to Pro and replace here with a hook?

**Comment by ihji** [comment]
Creating a hook in constant propagation module instead: https://github.com/semgrep/semgrep/pull/9180

**Pull request closed without merging**